improver_tests/calibration/test_QuantileMapping.py

```diff
@@ -0,0 +1,384 @@
+    same as values input in this case."""
+    _, quantiles = _build_empirical_cdf(simple_reference_array)
+    result = _inverted_cdf(simple_reference_array, quantiles)
+    np.testing.assert_array_equal(result, np.array([10, 20, 30, 40, 50]))
+
+
+def test__interpolated_inverted_cdf(simple_reference_array):
+    """Test _interpolated_inverted_cdf returns correct interpolated values."""
+    # Test with quantiles that fall between the reference data points
+    target_quantiles = np.array([0.3, 0.5, 0.7, 0.9])
+    result = _interpolated_inverted_cdf(simple_reference_array, target_quantiles)
+    # At quartile 0.3: interpolate between 0.2 (10) and 0.4 (20) -> 15
+    # At quartile 0.5: interpolate between 0.4 (20) and 0.6 (30) -> 25
+    # At quartile 0.7: interpolate between 0.6 (30) and 0.8 (40) -> 35
+    # At quartile 0.9: interpolate between 0.8 (40) and 1.0 (50) -> 45
+    expected = np.array([15, 25, 35, 45])
+    np.testing.assert_array_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "use_new_values, mapping_method, expected",
+    [
+        (False, "floor", np.array([10, 20, 30, 40, 50])),
+        (False, "interp", np.array([10, 20, 30, 40, 50])),
+        (True, "floor", np.array([20, 20, 30, 40, 50])),
+        (True, "interp", np.array([12.5, 22.5, 32.5, 42.5, 50])),
+    ],
+    ids=[
+        "same_values_to_map_floor",
+        "same_values_to_map_interp",
+        "different_values_to_map_floor",
+        "different_values_to_map_interp",
+    ],
+)
+def test_quantile_mapping(
+    simple_reference_array,
+    simple_forecast_array,
+    simple_new_values_to_map_array,
+    use_new_values,
+    mapping_method,
+    expected,
+):
+    values_to_map = (
+        simple_new_values_to_map_array if use_new_values else simple_forecast_array
+    )
+    result = quantile_mapping(
+        simple_reference_array,
+        simple_forecast_array,
+        values_to_map,
+        mapping_method=mapping_method,
+    )
+    np.testing.assert_array_equal(result, expected)
+
+
+def test_invalid_mapping_method_raises_error(
+    simple_reference_array, simple_forecast_array
+):
+    """Test that invalid mapping_method raises ValueError."""
+    with pytest.raises(ValueError, match="Unknown mapping method"):
+        quantile_mapping(
+            simple_reference_array, simple_forecast_array, mapping_method="kitten"
+        )
+
+
+@pytest.fixture
+def reference_cube():
+    """Fixture for creating a reference precipitation rate (mm/s) cube."""
+    data = np.array(
+        [
+            [
+                [2.63564289e-07, 8.47503543e-08, 3.35276127e-08],
+                [4.65661287e-08, 2.14204192e-08, 1.67638063e-08],
+                [8.38190317e-09, 1.21071935e-08, 2.23517418e-08],
+            ],
+            [
+                [5.58793545e-09, 3.81842256e-08, 2.03959644e-07],
+                [2.51457095e-08, 6.61239028e-08, 1.89989805e-07],
+                [5.49480319e-08, 9.40635800e-08, 1.64844096e-07],
```

Comment:
I think that potentially the clearest thing to do for all these fixtures would be define values firstly in mm/h, and then convert into m/s, and the mm/h inputs are more readable.